modeling-cmds/src/shared.rs

```diff
@@ -90,6 +100,28 @@ pub enum DistanceType {
     },
 }
 
+/// The type of origin
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case", tag = "type")]
+pub enum OriginType {
+    /// Local Origin (center of object bounding box).
+    Local {},
+    /// Global Origin (0, 0, 0).
+    Global {},
+    /// Custom Origin (user specified point).
+    Custom {
+        /// Custom origin point.
+        point: Point3d<f64>,
+    },
+}
+
+impl Default for OriginType {
+    /// Local Bounding Box Origin
+    fn default() -> Self {
```

Comment:
You can actually derive default: https://doc.rust-lang.org/std/default/trait.Default.html#enums -- I've added examples in the comments above.